sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/signaling/RealtimeNotificationConfigClient.java

```diff
@@ -106,6 +108,43 @@ public void onError(Throwable error) {
         return configResult[0];
     }
 
+    // Create HttpPipeline based on polices passed from Contonso. We only apply retry policy and user agent policy.
+    private HttpPipeline createHttpPipeline(HttpPipeline httpPipeline) {
+        List<HttpPipelinePolicy> customPolicies = getAllPolicies(httpPipeline);
+        // httpPipeline policies
+        HttpPipeline newPipeline = new HttpPipelineBuilder()
+            .httpClient(HttpClient.createDefault())
+            .policies(customPolicies.toArray(new HttpPipelinePolicy[0]))
+            .build();
+
+        return newPipeline;
+    }
+
+    /**
+     * Retrieves all policies from the given HttpPipeline instance.
+     *
+     * @param pipeline The HttpPipeline instance.
+     * @return A list of HttpPipelinePolicy instances.
+     */
+    public static List<HttpPipelinePolicy> getAllPolicies(HttpPipeline pipeline) {
```

Comment:
The functionality seems correct to me, but I would either rename this to something like `getEssentialPolicies()` or simply mesh both the above function and this one and leave it as `createHttpPipeline()`. The name `getAllPolicies()` implies that you're actually getting every single one of the policies in a given `HttpPipeline`.